examples/evm_bots.py

```diff
@@ -512,78 +542,84 @@ def log_and_show_block_info():
+    block_time_ = time.BlockTimeFP()
+    market, _, _ = sim_utils.get_initialized_hyperdrive_market_fp(pricing_model, block_time_, config)
+    return simulators.SimulatorFP(config, market, block_time_)
 
 
-def deploy_hyperdrive() -> ContractInstance:
+def deploy_hyperdrive(
+    config: simulators.ConfigFP, deployer: TestAccountAPI, base_contract: ContractInstance
+) -> ContractInstance:
     """Deploy Hyperdrive when operating on a fresh fork."""
     assert isinstance(deployer, TestAccountAPI)
-    initial_supply = int(config.target_liquidity * 1e18)
-    initial_apr = int(config.target_fixed_apr * 1e18)
-    initial_share_price = int(config.init_share_price * 1e18)
-    checkpoint_duration = 86400  # seconds = 1 day
-    checkpoints_per_term = 365
-    time_stretch = int(1 / simulator.market.time_stretch_constant * 1e18)
-    curve_fee = int(config.curve_fee_multiple * 1e18)
-    flat_fee = int(config.flat_fee_multiple * 1e18)
-    gov_fee = 0
-    base.mint(int(initial_supply * 1e18), sender=deployer)  # minted to sender
+    initial_supply = FixedPoint(config.target_liquidity, decimal_places=18)
+    initial_apr = FixedPoint(config.target_fixed_apr, decimal_places=18)
+    initial_share_price = FixedPoint(config.init_share_price, decimal_places=18)
+    checkpoint_duration = 86400  # seconds = 1 day # TODO: convert to FixedPoint after new spec is merged
+    checkpoints_per_term = 365  # TODO: convert to FixedPoint after new spec is merged
+    time_stretch = FixedPoint(FixedPoint("1.0") / simulator.market.time_stretch_constant, decimal_places=18)
+    curve_fee = FixedPoint(config.curve_fee_multiple, decimal_places=18)
+    flat_fee = FixedPoint(config.flat_fee_multiple, decimal_places=18)
+    gov_fee = FixedPoint(0)
+    base_contract.mint(initial_supply.int_value, sender=deployer)  # minted to sender
     # Deploy hyperdrive on the chain
-    hyperdrive_: ContractInstance = deployer.deploy(
+    hyperdrive: ContractInstance = deployer.deploy(
         project.get_contract("MockHyperdriveTestnet"),
-        base,
-        initial_apr,
-        initial_share_price,
-        checkpoints_per_term,
-        checkpoint_duration,
-        time_stretch,
-        (curve_fee, flat_fee, gov_fee),
+        base_contract,
+        initial_apr.int_value,
+        initial_share_price.int_value,
+        checkpoints_per_term,  # not in FixedPoint format
+        checkpoint_duration,  # not in FixedPoint format
+        time_stretch.int_value,
+        (curve_fee.int_value, flat_fee.int_value, gov_fee.int_value),
         deployer,
     )
     with ape.accounts.use_sender(deployer):
-        base.approve(hyperdrive_, initial_supply)
-        hyperdrive_.initialize(initial_supply, initial_apr, deployer, True)
-    return hyperdrive_
+        base_contract.approve(hyperdrive, initial_supply.int_value)
+        hyperdrive.initialize(initial_supply.int_value, initial_apr.int_value, deployer, True)
+    return hyperdrive
 
 
 if __name__ == "__main__":
```

Comment:
should we move this into a `main()` function since you already added passing variables between functions?

Reply:
Next pass-through, yes. I didn't carefully remove the global variable use, just as I noticed it. So we will need to go over the file again when we move stuff into elfpy.